apps/custom-config/wakeup-effect.js

```diff
@@ -202,28 +202,28 @@ class WakeupEffect extends BaseConfig {
    */
   onWakeupEffectStatusChangedFromIntent (action) {
     property.set('sys.wakeupwitch', action, 'persist')
-    if (action === 'close') {
-      this.notifyActivation([])
-    } else {
-      this.getFileList().then((fileList) => {
-        if (!fileList || !(fileList instanceof Array)) {
-          fileList = []
-        }
-        this.notifyActivation(fileList)
-      })
-    }
+    this.refresh()
   }
 
   /**
-   * init the activation status
+   * refresh the activation sound
    */
-  init () {
+  refresh () {
     var action = property.get('sys.wakeupwitch', 'persist')
     if (action === SWITCH_CLOSE) {
+      logger.info('wakeup effect turned off')
       this.notifyActivation([])
     } else {
       this.getFileList().then((fileList) => {
+        if (!fileList || !(fileList instanceof Array)) {
```

Comment:
use `Array.isArray` to determine if a value is an array.

Reply:
Updated.